### PR TITLE
[mono] Refactor UnmanagedCallersOnly_MonoAot tests

### PR DESCRIPTION
**Pull request by ivanpovazan**
This PR refactors the `UnmanagedCallersOnly_MonoAot` tests due to the reasons mentioned here: https://github.com/dotnet/runtime/issues/71532

### CODE REVIEW

**Comment by AaronRobinsonMSFT** [comment]
@ivanpovazan Thanks for working on this so quickly. I think we still want the test coverage, so it can't just be removed. I've added a suggestion on the issue for how to satisfy the confusion around test names and how to ensure we have the appropriate coverage.

**Comment by ivanpovazan** [comment]
@AaronRobinsonMSFT, marking this PR as draft so I can keep working on it based on your new suggestion.

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    public static int Main(string[] args)
+    {
+        var result = 100;
+
+        result += TestUnmanagedCallersOnlyValid_CallConvStdcall();
+        result += TestUnmanagedCallersOnlyValid_CallConvCdecl();
+
+        return result;
+    }
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:32]: Remove `Main` and place `[Fact]` on all test methods. We have as a source generator that will generate `Main` for you.

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        return DoubleImpl(n);
+    }
+
+    private static int DoubleImpl(int n)
+    {
+        return 2 * n;
+    }
+
+    public static int TestUnmanagedCallersOnlyValid_CallConvStdcall()
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:51]:
```suggestion
    [Fact]
    public static void TestUnmanagedCallersOnlyValid_CallConvStdcall()
```

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        Console.WriteLine($"Running {nameof(TestUnmanagedCallersOnlyValid_CallConvStdcall)}...");
+
+        int n = 12345;
+        int expected = DoubleImpl(n);
+        int actual = UnmanagedCallersOnlyDll.CallManagedProc_Stdcall(&ManagedDoubleCallback_Stdcall, n);
+
+        return expected == actual ? 0 : -1;
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:59]:
```suggestion
        Assert.Equal(expected, actual);
```

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    public static int TestUnmanagedCallersOnlyValid_CallConvCdecl()
+    {
+        Console.WriteLine($"Running {nameof(TestUnmanagedCallersOnlyValid_CallConvCdecl)}...");
+
+        int n = 12345;
+        int expected = DoubleImpl(n);
+        int actual = UnmanagedCallersOnlyDll.CallManagedProc_Cdecl(&ManagedDoubleCallback_Cdecl, n);
+
+        return expected == actual ? 0 : -1;
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:70]:
```suggestion
        Assert.Equal(expected, actual);
```

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        Console.WriteLine($"Running {nameof(TestUnmanagedCallersOnlyValid_CallConvStdcall)}...");
+
+        int n = 12345;
+        int expected = DoubleImpl(n);
+        int actual = UnmanagedCallersOnlyDll.CallManagedProc_Stdcall(&ManagedDoubleCallback_Stdcall, n);
+
+        return expected == actual ? 0 : -1;
+    }
+
+    public static int TestUnmanagedCallersOnlyValid_CallConvCdecl()
```
Comment (AaronRobinsonMSFT) [inline src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:62]:
```suggestion
    [Fact]
    public static void TestUnmanagedCallersOnlyValid_CallConvCdecl()
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
I meant to comment, not "Approve" yet.

**Comment by ivanpovazan** [comment]
> I meant to comment, not "Approve" yet.

No problem! I am still working on it - copying the tests and cleaning up the code (similar to what you have already suggested in your review)

File: src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs (hunk 1 above)
Comment (ivanpovazan) [reply src/tests/Interop/UnmanagedCallersOnlyBasic/UnmanagedCallersOnlyBasicTest.cs:32]: Do you want me to do the same for the `UnmanagedCallersOnlyTest`?

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Comment by AaronRobinsonMSFT** [comment]
Looks great @ivanpovazan. Thank you very much.

/cc @dotnet/interop-contrib

**Pull request merged**